src/containers/Search.jsx

```diff
@@ -1,208 +1,312 @@
-                    <div style={{ width: `100%`, height: `100%` }} />
-                  }
-                  myPlaces={[
-                    { id: "1", pos: { lat, lng } },
-                    {
-                      id: "2",
-                      pos: { lat: selectedLat, lng: selectedLng },
-                    },
-                  ]}
-                />
-              </div>
-            </div>
-          )}
-        </div>
-      </>
-    );
-  }
+	setNewLocation = (lat, lng) => {
+		/**
+		 * Set Latitude and Logitude of selected hospital
+		 * @event setNewLocation
+		 * @type {updater}
+		 * @property {selectedLat} state.selectedLat
+		 * @property {selectedLng} state.selectedLng
+		 */
+		this.setState({
+			selectedLat: lat,
+			selectedLng: lng,
+		});
+	};
+
+	/**
+	 * Find distance between current location and selected hospital
+   
+	 * @param {number} lat1 - Latitude of the source location.
+	 * @param {number} lng1 - Longitude of the source location.  
+	 * @param {number} lat2 - Latitude of the destination location.
+	 * @param {number} lng2 - Longitude of the destination location.  
+   * @param {string} unit - required unit of distance in kilometers('K') or nautical miles('N').
+   * @return {number} The distance between two co-ordinates
+	 */
+	generateDistance = (lat1, lon1, lat2, lon2, unit) => {
+		if (lat1 === lat2 && lon1 === lon2) {
+			return 0;
+		} else {
+			var radlat1 = (Math.PI * lat1) / 180;
+			var radlat2 = (Math.PI * lat2) / 180;
+			var theta = lon1 - lon2;
+			var radtheta = (Math.PI * theta) / 180;
+			var dist =
+				Math.sin(radlat1) * Math.sin(radlat2) +
+				Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta);
+			if (dist > 1) {
+				dist = 1;
+			}
+			dist = Math.acos(dist);
+			dist = (dist * 180) / Math.PI;
+			dist = dist * 60 * 1.1515;
+			if (unit === 'K') {
+				dist = dist * 1.609344;
+			}
+			if (unit === 'N') {
+				dist = dist * 0.8684;
+			}
+			return dist.toFixed(2);
+		}
+	};
+
+	/**
+	 * Deeplink to Uber app or store based on app installation status
+   
+	 * @param {string} address - address of the selected location.
+	 * @param {number} lat - Latitude of the selected location.
+	 * @param {number} lng - Longitude of the selected location.
+	 */
+	redirectToUber = (address, lat, lng) => {
+		const { isMobile } = this.state;
+		if (isMobile) {
+			window.location = `https://m.uber.com/ul/?action=setPickup&client_id=${UBER_CLIENT_ID}&pickup=my_location&dropoff[formatted_address]=${encodeURI(
+				address,
+			)}&dropoff[latitude]=${lat}&dropoff[longitude]=${lng}`;
+		}
+	};
+
+	/**
+	 * A react lifecycle method called when the component has mounted.
+	 * It calls the getAddressFromLocation method right after updating.
+	 */
+	componentDidMount() {
+		this.getAddressFromLocation();
+	}
+
+	/**
+	 * The render function.
+	 */
+	render() {
+		const {
+			lat,
+			lng,
+			city,
+			results,
+			isLoading,
+			selectedLat,
+			selectedLng,
+			isMobile,
+			fromAddress,
```

Comment:
fromAddress is used?

Reply:
I fixed these issue, is fetching the right city name from geolocation in '/search' fixed?  I see it's fixed in https://dreamy-keller-b8ad36.netlify.app . Needs to be taken from result_type. Want to confirm before proceeding, don't want to redo the same fix.

Reply:
@nivedhasamy just cross check. it should. that link is the main test link that goes live so yeah it should just show city; 
Take a git pull and merge from master and check on ur local;